scio-examples/src/main/scala/com/spotify/scio/examples/extra/BinaryInOut.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2023 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.examples.extra
+
+import com.spotify.scio.ContextAndArgs
+import java.nio.ByteBuffer
+import com.spotify.scio.io.BinaryIO.BinaryFileReader
+import java.io.InputStream
+
+// Example: Binary Input and Output
+// Usage:
+// `sbt "runMain com.spotify.scio.examples.extra.BinaryInOut
+// --project=[PROJECT] --runner=DataflowRunner --region=[REGION NAME] --output=[OUTPUT]"`
+
+object BinaryInOut {
```

Comment:
great example!

Can we also add a test for this? I feel like it's common for users to be confused on how to use `JobTest` with new IOs so it would be great if we had an example test to point them to.

Reply:
I don't think this would be useful; the logic is encapsulated in the reader, so all the test would do is assert that some array values are written and read back, but it wouldn't capture (or test) any of the file parsing or formatting.